aFuzzyBear/grateful-gravity
# Implement jQuery Version Detection Feature

This pull request introduces a new feature to the Check 4 jQuery application that allows users to query various URLs to detect and report the version of jQuery used by those sites. The feature leverages Puppeteer for headless browser interaction, providing a seamless and efficient way to analyze web pages for jQuery usage. This update aims to enhance the application's utility for developers and researchers interested in web technology stacks.

## Changes:

New Feature: Added a utility function in utils/getVersion.ts to detect the jQuery version from the given URL.
Enhancements: Improved URL sanitization in utils/normaliseURL.ts to handle a wider range of input variations and ensure the application can process URLs more reliably.
Documentation: Updated the README.md to include detailed information about the new feature, how to use it, and expanded the project structure description for better clarity.
Bug Fixes: Addressed an issue where certain types of modals on web pages could prevent the application from correctly detecting jQuery.

## How to Test:
Ensure all dependencies are installed by running npm install.
Start the local development server with npm run dev.
Navigate to the homepage and submit URLs through the provided form to test the jQuery version detection feature.
For testing purposes, use the recommended URLs provided in the README.md, including edge cases like https://cpanel.net/ where jQuery detection is challenging.

## Known Issues:
Some web pages with complex JavaScript execution or heavy use of iframes may still pose challenges for accurate jQuery version detection. Future updates will aim to improve coverage and accuracy.

## Future Work:
Expand the utility to detect additional JavaScript libraries and frameworks.
Implement a more robust handling mechanism for web pages with dynamic content and modals.
